hpx/lcos/local/recursive_mutex.hpp

```diff
@@ -35,7 +35,7 @@ namespace hpx { namespace lcos { namespace local
 
         private:
             std::atomic<std::uint64_t> recursion_count;
-            std::atomic<hpx::execution::agent> locking_context;
+            std::atomic<hpx::execution::agent_ref> locking_context;
```

Comment:
Is it ok to use a non-POD type with `std::atomic`?

Reply:
Yes, that's fine, `agent_ref` should provide the necessary properties as described here: https://en.cppreference.com/w/cpp/atomic/atomic (under Primary template)